hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionReplicaManager.java

```diff
@@ -190,57 +190,76 @@ Address checkAndGetPrimaryReplicaOwner(int partitionId, int replicaIndex) {
      * was not removed while the cluster was not active. Also cancel any currently scheduled sync requests for the given
      * partition and schedule a new sync request that is to be run in the case of timeout
      */
-    private void sendSyncReplicaRequest(int partitionId, Collection<ServiceNamespace> syncNamespaces,
+    private void sendSyncReplicaRequest(int partitionId, Collection<ServiceNamespace> requestedNamespaces,
             int replicaIndex, Address target) {
         if (node.clusterService.isMemberRemovedInNotJoinableState(target)) {
             return;
         }
 
-        if (!tryToAcquireReplicaSyncPermit()) {
+        int permits = tryAcquireReplicaSyncPermits(requestedNamespaces.size());
+        if (permits == 0) {
```

Comment:
I'm guessing the rest of the namespaces are synchronized the next time the inconsistency is detected (as it was before when there were no permits available), right?

Reply:
Yes, that's right.